helix-core/src/main/java/org/apache/helix/task/WorkflowContext.java

```diff
@@ -228,10 +265,22 @@ public long getLastJobPurgeTime() {
   }
 
   public void setName(String name) {
-    _record.setSimpleField(WorkflowContextProperties.NAME.name(), name);
+    String oldName = getName();
+    if (oldName == null || !oldName.equals(name)) {
```

Comment:
Those checks could in general be consolidated to `!name.equals(oldName)`.

Reply:
Done.